File: cpp/01.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <chrono>

using namespace std;
using namespace std::chrono;

// Function to generate random numbers
vector<int> generateRandomNumbers(int size) {
    vector<int> nums(size);
    srand(time(0));
    for (int i = 0; i < size; ++i) {
        nums[i] = rand();
    }
    return nums;
}

// Function to perform benchmarking
void benchmarkSortingAlgorithms(int size) {
    vector<int> nums = generateRandomNumbers(size);
    vector<int> nums_copy = nums;

    // Benchmarking selection sort
    auto start = high_resolution_clock::now();
    sort(nums.begin(), nums.end());
    auto stop = high_resolution_clock::now();
    auto duration_selection_sort = duration_cast<milliseconds>(stop - start).count();
    cout << "Selection Sort Time: " << duration_selection_sort << " milliseconds" << endl;

    // Benchmarking C++ sort function (introsort)
    start = high_resolution_clock::now();
    sort(nums_copy.begin(), nums_copy.end());
    stop = high_resolution_clock::now();
    auto duration_cpp_sort = duration_cast<milliseconds>(stop - start).count();
    cout << "C++ Sort (introsort) Time: " << duration_cpp_sort << " milliseconds" << endl;
}

int main() {
    int size;
    cout << "Enter the size of the array: ";
    cin >> size;

    benchmarkSortingAlgorithms(size);
    return 0;
}

File: python/01.py
import random
import time

# Function to generate random numbers
def generate_random_numbers(size):
    return [random.randint(0, 1000) for _ in range(size)]

# Function to perform benchmarking
def benchmark_sorting_algorithms(size):
    nums = generate_random_numbers(size)
    nums_copy = nums.copy()

    # Benchmarking selection sort
    start = time.time()
    nums.sort()
    duration_selection_sort = (time.time() - start) * 1000
    print(f"Selection Sort Time: {duration_selection_sort} milliseconds")

    # Benchmarking Python's built-in sort function (Timsort)
    start = time.time()
    nums_copy.sort()
    duration_python_sort = (time.time() - start) * 1000
    print(f"Python Sort (Timsort) Time: {duration_python_sort} milliseconds")

if __name__ == "__main__":
    size = int(input("Enter the size of the list: "))
    benchmark_sorting_algorithms(size)
